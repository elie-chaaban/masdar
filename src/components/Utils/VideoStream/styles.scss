.player-wrapper {
  position: relative;
  height: 100px;

  .react-player {
    position: absolute;
    top: 0;
    left: 0;
  }
}
.player-wrapper-no-hight {
  position: relative;

  .react-player {
    position: absolute;
    top: 0;
    left: 0;
  }
}
video:focus {
  outline: none;
}
