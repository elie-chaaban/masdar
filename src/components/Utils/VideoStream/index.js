import React from 'react';
import ReactPlayer from 'react-player';
import PropTypes from 'prop-types';

import './styles.scss';

const VideoStream = ({streamUrl, wrapperClassName}) => {
  return (
    <div className={wrapperClassName ? wrapperClassName : 'player-wrapper'}>
      <ReactPlayer url={streamUrl} playing width="100%" height="100%" controls={true} pip={true} />
    </div>
  );
};

VideoStream.propTypes = {
  streamUrl: PropTypes.string.isRequired
};
export default React.memo(VideoStream);
