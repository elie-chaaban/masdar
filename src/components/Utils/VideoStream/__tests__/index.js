import React from 'react';
import {shallow} from 'enzyme';
import VideoStream from '..';

test('video Stream', () => {
  const dom = shallow(<VideoStream streamUrl="url" />);
  expect(dom).toMatchSnapshot();
  expect(dom.find('ReactPlayer').prop('url')).toBe('url');
});
