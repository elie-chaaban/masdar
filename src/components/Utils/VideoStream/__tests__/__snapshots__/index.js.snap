// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`video Stream 1`] = `
<div
  className="player-wrapper"
>
  <ReactPlayer
    config={
      Object {
        "dailymotion": Object {
          "params": Object {
            "api": 1,
            "endscreen-enable": false,
          },
        },
        "facebook": Object {
          "appId": "1309697205772819",
          "playerId": null,
          "version": "v3.3",
        },
        "file": Object {
          "attributes": Object {},
          "dashVersion": "2.9.2",
          "flvVersion": "1.5.0",
          "forceAudio": false,
          "forceDASH": false,
          "forceFLV": false,
          "forceHLS": false,
          "forceVideo": false,
          "hlsOptions": Object {},
          "hlsVersion": "0.13.1",
          "tracks": Array [],
        },
        "mixcloud": Object {
          "options": Object {
            "hide_cover": 1,
          },
        },
        "soundcloud": Object {
          "options": Object {
            "buying": false,
            "download": false,
            "liking": false,
            "sharing": false,
            "show_comments": false,
            "show_playcount": false,
            "visual": true,
          },
        },
        "twitch": Object {
          "options": Object {},
          "playerId": null,
        },
        "vidyard": Object {
          "options": Object {},
        },
        "vimeo": Object {
          "playerOptions": Object {
            "autopause": false,
            "byline": false,
            "portrait": false,
            "title": false,
          },
        },
        "wistia": Object {
          "options": Object {},
          "playerId": null,
        },
        "youtube": Object {
          "embedOptions": Object {},
          "onUnstarted": [Function],
          "playerVars": Object {
            "iv_load_policy": 3,
            "modestbranding": 1,
            "playsinline": 1,
            "rel": 0,
            "showinfo": 0,
          },
        },
      }
    }
    controls={true}
    height="100%"
    light={false}
    loop={false}
    muted={false}
    onBuffer={[Function]}
    onBufferEnd={[Function]}
    onDisablePIP={[Function]}
    onDuration={[Function]}
    onEnablePIP={[Function]}
    onEnded={[Function]}
    onError={[Function]}
    onPause={[Function]}
    onPlay={[Function]}
    onProgress={[Function]}
    onReady={[Function]}
    onSeek={[Function]}
    onStart={[Function]}
    pip={true}
    playbackRate={1}
    playing={true}
    playsinline={false}
    progressInterval={1000}
    style={Object {}}
    url="url"
    volume={null}
    width="100%"
    wrapper="div"
  />
</div>
`;
